# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/chrome-webidl/ClonedErrorHolder.webidl
# Commit: e4783542b7c8
# Full Hash: e4783542b7c854ae4aa5cf60ac27e5e1ff19d56a
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2019-12-07 21:55:34
# Regressor Bug: 1588839
# File Overlap Count: 1
# Description:
#   Bug 1588839 - Part 1 - Add helper to allow structured cloning Error/Exception objects. r=bzbarsky
#   
#   In order to be able to reasonably debug error results from things like
#   JSWindowActor.sendQuery, we need to be able to clone errors across process
#   boundaries, so that they can be propagated to the caller that initiated a
# ==============================================================================

diff -r df371f905d3a -r e4783542b7c8 dom/chrome-webidl/ClonedErrorHolder.webidl
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/chrome-webidl/ClonedErrorHolder.webidl	Sat Dec 07 18:59:14 2019 +0000
@@ -0,0 +1,18 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A stub object to hold an Error, Exception, or DOMException object to be
+ * transferred via structured clone. The object will automatically be decoded
+ * directly into the type of the error object that it wraps.
+ *
+ * This is a temporary workaround for lack of native Error and Exception
+ * cloning support, and can be removed once bug 1556604 and bug 1561357 are
+ * fixed.
+ */
+[ChromeOnly, Exposed=Window]
+interface ClonedErrorHolder {
+  [Throws]
+  constructor(object aError);
+};